### PR TITLE
Armor tweaks

### PR DESCRIPTION
**Pull request by TechLord22**
Improves the code of all MetaArmors. Vastly improves jetpack behavior to use SimplyJetpacks' implementation. Fixes inventory charging outright not working. Improves recipes for armor. Improves armor lang.

### CODE REVIEW

**Comment by Syrcan** [comment]
Issues found so far.

**Minor (might not need fix):**
Flight cannot be toggled for the lower tier jetpacks.
Opening inventory keeps the currently pressed movement keys active.
Fueled Jetpack cannot be equipped with right-click.
Fueled Jetpack internal tank represents a ridiculous amount of energy.
Protection level in unclear, as the armor protects better than the armor points it provides.
(Recipe) Reduce Field Generators to one per QuarkTech(tm) piece of equipment.

**Major (needs fix in my opinon):**
Advanced Nano Chestplate recipe is available at HV despite being EV-tier. Just make it HV-tier.
All Chestplates consume electricty when landing on flat ground (regardless of fall damage).
Only the MV Jetpack consumes electric when flying.
Pressing Sneak in Hover Mode does not descend.
The fall damage fix has not been applied to the Fueled Jetpack.
Fueled Jetpack accepts some fuels that it refuses to burn (Methanol is an example).
Fueled Jetpack appears to consume once ever, then nothing.
"Supply Mode" tooltip is duplicated.
Supply Mode has a will on its own (does not always work).
BIG BAD CME from Supply Mode trying to charge Fueled Jetpack.

**Comment by ALongStringOfNumbers** [comment]
Duplicate "Supply Mode" tooltips

**Comment by Syrcan** [comment]
An update on the feedback, again sorted with minor and major.

**Minor:**
What I meant with armor was *not* the text in the tooltip. It was the fact that the armor seems to protect better than the armor points it gives in your armor bar.
"Supply Mode" tooltip is still duplicated for the adv nano.

**Major:**
Emergency Hover Mode is more annoying than useful (unless it can be turned off?).
Adv Nano is still EV-tier. It should be HV.
Sneak in Hover Mode should allow the player to descend faster, else it you have to spam the hotkey for precise movement.
Titanium in the assembler recipe is megacursed (jk I love it)

Other fixes seem to work, that's nice!

File: src/main/java/gregtech/common/items/MetaItem1.java [hunk 1]
```diff
@@ -568,8 +568,8 @@ public void registerSubItems() {
         ZERO_POINT_MODULE = addItem(752, "zpm").addComponents(ElectricStats.createBattery(2000000000000L, GTValues.ZPM, true)).setModelAmount(8);
         ULTIMATE_BATTERY = addItem(753, "max.battery").addComponents(ElectricStats.createRechargeableBattery(Long.MAX_VALUE, GTValues.MAX)).setUnificationData(OrePrefix.battery, Tier.Maximum).setModelAmount(8);
 
-        IMPELLER_MV = addItem(776, "impeller.mv").setRarity(EnumRarity.UNCOMMON);
-        IMPELLER_HV = addItem(777, "impeller.hv").setRarity(EnumRarity.RARE);
+        POWER_THRUSTER = addItem(776, "impeller.mv").setRarity(EnumRarity.UNCOMMON);
+        POWER_THRUSTER_ADVANCED = addItem(777, "impeller.hv").setRarity(EnumRarity.RARE);
```
Comment (ALongStringOfNumbers) [inline src/main/java/gregtech/common/items/MetaItem1.java:572]: If you are going to change these might as well change the item names as well

**Review by ALongStringOfNumbers — COMMENTED** [review]
The "Hover Mode" and "Fly Mode" helpers in the armor information display don't display when the player first puts on the armor. Only toggling these for the first time will make the helper information appear

File: src/main/java/gregtech/common/items/armor/NanoMuscleSuite.java [hunk 2]
```diff
@@ -18,72 +18,68 @@
...
-                PotionEffect night_vision = player.getActivePotionEffect(MobEffects.NIGHT_VISION);
-                if (blindness != null) {
-                    if (blindness.getDuration() < 1) player.removePotionEffect(MobEffects.BLINDNESS);
-                }
-                if (night_vision != null) {
-                    if (night_vision.getDuration() < 1) player.removePotionEffect(MobEffects.NIGHT_VISION);
-                }
+            if (nightvision && !world.isRemote && item.getCharge() >= 4) {
+                player.removePotionEffect(MobEffects.BLINDNESS);
+                player.addPotionEffect(new PotionEffect(MobEffects.NIGHT_VISION, 999999, 0, true, false));
```
Comment (htmlcsjs) [inline src/main/java/gregtech/common/items/armor/NanoMuscleSuite.java:65]: This doesnt account for if you take off the helmet

File: src/main/java/gregtech/common/items/armor/QuarkTechSuite.java [hunk 3]
```diff
@@ -45,175 +47,175 @@ public QuarkTechSuite(EntityEquipmentSlot slot, int energyPerUse, long capacity,
...
-                    if (skylight > 8) {
-                        player.removePotionEffect(MobEffects.NIGHT_VISION);
-                        player.addPotionEffect(new PotionEffect(MobEffects.BLINDNESS, 100, 0, true, true));
-                    } else {
-                        player.removePotionEffect(MobEffects.BLINDNESS);
-                        player.addPotionEffect(new PotionEffect(MobEffects.NIGHT_VISION, 300, 0, true, true));
-                    }
+            if (nightvision && !world.isRemote && item.getCharge() >= energyPerUse) {
+                player.removePotionEffect(MobEffects.BLINDNESS);
+                player.addPotionEffect(new PotionEffect(MobEffects.NIGHT_VISION, 999999, 0, true, false));
```
Comment (htmlcsjs) [inline src/main/java/gregtech/common/items/armor/QuarkTechSuite.java:124]: This doesnt account for if you take off the helmet

File: src/main/java/gregtech/common/items/armor/NightvisionGoggles.java [hunk 4]
```diff
@@ -61,35 +56,23 @@ public void onArmorTick(World world, EntityPlayer player, ItemStack itemStack) {
...
-                    player.addPotionEffect(new PotionEffect(MobEffects.BLINDNESS, 999999, 0, true, true));
-                } else {
-                    player.removePotionEffect(MobEffects.BLINDNESS);
-                    player.addPotionEffect(new PotionEffect(MobEffects.NIGHT_VISION, 999999, 0, true, true));
-                }
-                ret = true;
-                item.discharge((energyPerUse / 100), this.tier, true, false, false);
+            if (nightvision && !world.isRemote && item.getCharge() >= energyPerUse) {
+                player.removePotionEffect(MobEffects.BLINDNESS);
+                player.addPotionEffect(new PotionEffect(MobEffects.NIGHT_VISION, 999999, 0, true, false));
```
Comment (htmlcsjs) [inline src/main/java/gregtech/common/items/armor/NightvisionGoggles.java:61]: This doesnt account for if you take off the goggles

**Review by htmlcsjs — COMMENTED** [review]
You might want to add some tooltips that show what each armour piece does like step assist or nyooming

**Comment by ALongStringOfNumbers** [comment]
Having a separate fly mode key makes the armor feel really clunky, especially since you can turn off fly mode, but keep on hover mode, which seems like to me should keep the player hovering but does not.

**Review by DStrand1 — APPROVED** [review]

**Pull request merged**